HW3/unit_tests/test_client.py

```diff
@@ -1,95 +1,74 @@
 import os.path
 import unittest
-from unittest.mock import patch  # имитация действия функции sys запуск из терминала
 import sys
-import client
-
-sys.path.append(os.path.join(os.getcwd(), '..'))
-
-
-
-class TestCheckPort(unittest.TestCase):
-
-    @patch.object(sys, 'argv', ['client.py', '-p'])
-    def test_without_port_with_p(self):
-        self.assertRaises(IndexError, client.check_port)
-
-    @patch.object(sys, 'argv', ['client.py', '-p', 777])
-    def test_with_wrong_port_min(self):
-        self.assertRaises(ValueError, client.check_port)
-
-    @patch.object(sys, 'argv', ['client.py', '-p', 98777])
-    def test_with_wrong_port_max(self):
-        self.assertRaises(ValueError, client.check_port)
-
-    @patch.object(sys, 'argv', ['client.py', '-p', 'f45454'])
-    def test_with_wrong_type_port_str(self):
-        self.assertRaises(ValueError, client.check_port)
-
-    @patch.object(sys, 'argv', ['client.py', '-p', 4.5454])
-    def test_with_wrong_type_port_float(self):
-        self.assertRaises(ValueError, client.check_port)
-
-
-class TestCheckAddress(unittest.TestCase):
-
-    def test_with_a_without_address(self):
-        with patch.object(sys, 'argv', ['client.py', '-a']):
-            self.assertRaises(IndexError, client.check_address)
+from client import create_presence, process_answer
 
 
-class TestValidateAddress(unittest.TestCase):
-
-    def test_wrong_delimetr_address(self):
-        self.assertRaises(TypeError, client.validate_address, '127,0,0.1')
-
-    def test_wrong_length_address_min(self):
-        self.assertRaises(TypeError, client.validate_address, '401.0.1')
-
-    def test_wrong_length_address_max(self):
-        self.assertRaises(TypeError, client.validate_address, '401.0.1.2.5')
-
-    def test_wrong_address(self):
-        self.assertRaises(ValueError, client.validate_address, '401.0.0.1')
+sys.path.append(os.path.join(os.getcwd(), '..'))
```

Comment:
Changing the sys.path list must be BEFORE the import. Otherwise it doesn't make sense. 
first
```sys.path.append(os.path.join(os.getcwd(), '..'))```
and ONLY next
``` from client import create_presence, process_answer```

Reply:
Hi!
now I finally completed it with import, it did not give me an error, because the Sources Root was indicated on the project folder.
Thanks!